libflux/src/ast/walk/mod.rs

```diff
@@ -0,0 +1,363 @@
+#[cfg(test)]
+mod tests;
+
+use crate::ast::*;
+use std::cell::{RefCell, RefMut};
+use std::rc::Rc;
+
+// Node represents any structure that can appear in the AST
+#[derive(Debug)]
+pub enum Node<'a> {
```

Comment:
This seems like it should be in the `ast` module.

Reply:
It is not needed in the `ast` module and is only used here so I thought it best to just leave it here. The `ast` doesn't need a generic concept of a `Node` because the use of enums means that we can always be specific about what ast nodes can go where. There is never a branch of the AST were all nodes are possible.

Reply:
Sure. We'll refactor later if this isn't the case. I approved so we're good.